src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx

```diff
@@ -106,4 +133,46 @@ describe('useSidebarMetadataFetcher', () => {
             }),
         );
     });
+
+    test('should handle metadata instance removal', async () => {
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({ templateInstances: mockTemplateInstances, templates: mockTemplates });
+        });
+        mockAPI.deleteMetadata.mockImplementation((file, template, successCallback) => {
+            successCallback(mockTemplateInstances[0]);
+        });
+
+        const { result } = setupHook();
+        await waitFor(() => expect(result.current.templateInstances).toEqual(mockTemplateInstances));
```

Comment:
we dont need the `await waitFor` on any of the `expects`. on the functions like when you do this `await waitFor(() => result.current.deleteMetadataInstance(mockTemplateInstances[0]));` is still required in these cases.